Sources/SwifterSwift/SwiftStdlib/KeyedDecodingContainer.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  KeyedDecodingContainer.swift
+//  SwifterSwift
+//
+//  Created by Francesco Deliro on 23/10/2019.
+//  Copyright © 2019 SwifterSwift
+//
+
+#if canImport(Foundation)
+import Foundation
+#endif
+
+public extension KeyedDecodingContainer {
+
+    #if canImport(Foundation)
+    /// SwifterSwift: Try to decode a Bool as Int then String before decoding as Bool.
+    ///
+    /// - Parameter key: Key.
+    /// - Returns: Decoded Bool value.
+    /// - Throws: Decoding error.
+    func decodeBoolAsIntOrString(forKey key: Key) throws -> Bool {
+        if let intValue = try? decode(Int.self, forKey: key) {
+            return (intValue as NSNumber).boolValue
+        } else if let stringValue = try? decode(String.self, forKey: key) {
+            return (stringValue as NSString).boolValue
+        } else {
+            return try decode(Bool.self, forKey: key)
+        }
+    }
+    #endif
+
+    #if canImport(Foundation)
+    /// SwifterSwift: Try to decode a Bool as Int then String before decoding as Bool if present.
+    ///
+    /// - Parameter key: Key.
+    /// - Returns: Decoded Bool value.
+    /// - Throws: Decoding error.
+    func decodeBoolAsIntOrStringIfPresent(forKey key: Key) throws -> Bool? {
+        if let intValue = try? decodeIfPresent(Int.self, forKey: key) {
+            return (intValue as NSNumber).boolValue
+        } else if let stringValue = try? decodeIfPresent(String.self, forKey: key) {
+            return (stringValue as NSString).boolValue
```

Comment:
Please replace this with `return stringValue == "1"`

Reply:
Quoting Apple doc:

**NSString boolValue**
> This property is true on encountering one of "Y", "y", "T", "t", or a digit 1-9—the method ignores any trailing characters. This property is false if the receiver doesn’t begin with a valid decimal text representation of a number.

> The property assumes a decimal representation and skips whitespace at the beginning of the string. It also skips initial whitespace characters, or optional -/+ sign followed by zeroes.